### ISSUE TITLE
Explore different method of triggering Browsersync reload

### ISSUE THREAD

**Issue by erikjung** [cloudfour/core-gulp-tasks#18]
Rather than using the `files` option for Browsersync reloads, see if we can just handle watch tasks on a per `gulpfile.js` basis, and call `bs.reload()` or `bs.stream()` when needed.

**Comment by tylersticka** [comment]
I've noticed in our most recent project utilizing this module that I _am_ seeing CSS injections occur where they weren't before. I suspect that perhaps BrowserSync introduced an update that intelligently injects if CSS files are what change?

There's still the possibility that performance might be improved if there is only one process watching files (rather than Gulp watching source files then BrowserSync watching compile files), but the _feature_ complaint I have (regarding streaming of CSS files) may be a moot point now. (Though I'd still like one other person to test whether or not I'm going crazy in terms of seeing CSS injection happening as-is.)

**Comment by erikjung** [comment]
@tylersticka I was going to ask about that, because I too saw the BS notifications saying `injected foo.css`. Is there any way we can confirm that this is the same as manually calling `bs.stream()`?

**Comment by tylersticka** [comment]
@erikjung From [the `files` option section of their docs](http://www.browsersync.io/docs/options/#option-files), emphasis mine:

> Changes you make will either be **injected into the page (CSS & images)** or will cause all browsers to do a full-page refresh.

Those updates were introduced in `2.6.0`, so we should also be experiencing those benefits in our internal redesign project already.

Looking at our RFD project, it's clear why we weren't getting this benefit... presumably because we were grafting BrowserSync onto Divshot's server tool, the BS reload was happening like this:

```js
gulp.watch(config.watch.dest, bs.reload);
```

To summarize, it seems there are two ways to get BrowserSync's nifty file-streaming:

1. Call the `.reload` or `.stream` methods from the relevant Gulp tasks individually.
2. Use the `files` option to let BrowserSync handle reloading independently of Gulp.

![](http://bukk.it/themoreyouknow.gif)

**Comment by erikjung** [comment]
@tylersticka I'll leave this open, but won't prioritize it. We can always opt out of using this core `serve` task on a per-project basis and implement our own version that provides more flexibility with the Browsersync instance in use. The core task basically just calls `browsersync.init(options)`, so we don't really have to reinvent much if we decide not to use it :)

**Comment by tylersticka** [comment]
@erikjung That's completely fair. At this point it's more of a semantic argument than anything, and those should have subterranean priority.
